### ISSUE TITLE
Stale container images problematic

### ISSUE THREAD

**Issue by dboreham** [cerc-io/stack-orchestrator#262]
I started work on a laptop I haven't used for a few weeks and ran into several strange container build errors that turned out to be due to the docker registry on the machine having old versions of base images. We should add something in stack orchestrator to manage this problem, ideally not requiring hashes of every single reference to a base image everywhere (e.g. some of our transitive dependencies have Dockerfiles that we don't want to have to fork).

**Comment by AFDudley** [comment]
FWIW, It seems moderately likely to me that we'll need to store the transitive dependencies in the registry.

**Comment by dboreham** [comment]
Definitely. This is more about the pre-problem of identifying the base images among a big set of containers we build (automatically, not by having a human review them).

We can use the registry to store that metadata and write a tool to update those records when base image updates are published.

Really the docker tooling should do this, but it turns out it was designed to feed you total mystery meat.
